mp/src/vgui2/vgui_controls/CvarTextEntry.cpp

```diff
@@ -62,6 +62,22 @@ void CvarTextEntry::GetSettings(KeyValues* pOutResource)
     pOutResource->SetString("cvar_name", m_cvarRef.GetName());
 }
 
+void CvarTextEntry::SetText(const char* text)
+{
+    if (text[strnlen(text, MAX_CVAR_TEXT) - 1] == '.')
```

Comment:
What happens when `text` is empty?

Reply:
Which I would assume happens when the user selects all text in the entry and removes it, especially for cvars that accept any string value

Reply:
> 
> 
> What happens when `text` is empty (`""`)?

Currently, [`SetText` is never called with an empty string](https://github.com/momentum-mod/game/blob/af1cf4302adea52afa963db43e17d27ed5e5f9b3/mp/src/vgui2/vgui_controls/CvarTextEntry.cpp#L99). Could add a guard in the case it is explicitly called outside of the class. 

> 
> 
> Which I would assume happens when the user selects all text in the entry and removes it, especially for cvars that accept any string value

Other methods in the baseclass are called when the text is deleted via backspace or delete. I believe these are, funnily enough, `Backspace()` and `Delete()`. There is also a guard against setting the cvar when the text is empty [here](https://github.com/momentum-mod/game/blob/af1cf4302adea52afa963db43e17d27ed5e5f9b3/mp/src/vgui2/vgui_controls/CvarTextEntry.cpp#L85). This works when selecting the text and replacing it with another value as well.

There is the case where the panel is closed with no text in the textentry, then opened. The textentry is still empty in this case but it should populate with the current cvar value. I'll fix that in another commit.

Reply:
Added a guard in case the `SetText` override ride is called with an empty string.

Changed the `OnThink()` method to call `Reset()` when the text entry is not in focus. Clicking off the text entry, switching tabs, or exiting/reopening the panel will now reset it to the current value of its corresponding cvar.